### ISSUE TITLE
`split`: does not support any `--number=CHUNKS` option except `N`

### ISSUE THREAD

**Issue by ndd7xv** [uutils/coreutils#3201]
From GNU's man page on `split`:

```
-n, --number=CHUNKS
              generate CHUNKS output files; see explanation below

...

CHUNKS may be:
       N      split into N files based on size of input

       K/N    output Kth of N to stdout

       l/N    split into N files without splitting lines/records

       l/K/N  output Kth of N to stdout without splitting lines/records

       r/N    like 'l' but use round robin distribution

       r/K/N  likewise but only output Kth of N to stdout
```

Currently, coreutil's `split` only supports the `--number=N` option. The applicable `--number=CHUNKS` options should respect the specified separator that determines the records (if this is being worked on after https://github.com/uutils/coreutils/issues/3192).

**Comment by jfinkels** [comment]
Yeah, I have these all implemented on my local machine, but I was waiting on pull request #2980 to be merged.

**Comment by andrewbaptist** [comment]
@jfinkels do you still have the changes for this implemented - I was trying to get all the split tests passing and didn't want to redo any work if you already have this done.

**Comment by jfinkels** [comment]
No, sorry, it's been too long. Go for it

**Comment by zhitkoff** [comment]
this can be closed as PR #5252 implemented missing functionality for --number option
